### DISCUSSION TITLE
Cannot access Sigenstor from homeassistant

### DISCUSSION THREAD

**Discussion by wigglememore** [TypQxQ/Sigenergy-Home-Assistant-Integration#34] [Q&A] [answered]
Hi all,

Porting this questions over to a discussion from the intro thread.

I have a Sigenstor 8.0 SP, 8kWh battery, backup gateway and (soon) an AC charger. I have configured the integration in my Home Assistant instance however everything is showing as unavailable, so I'm not receiving any information. I have a screenshot from my installer showing the modbus server is active. 
![2024-11-08 16-56-56](https://github.com/user-attachments/assets/c8c1987e-3d7a-4937-8adc-d0d0712d1e04)

I have two devices showing up in my Unifi controller but neither of the IPs seem to work.
![384523231-8bfbaf91-a90a-4eb0-accf-bc2fcec4cf5f](https://github.com/user-attachments/assets/69030ceb-af7c-4daf-ab0a-068954324c8f)

I'm trying to figure out why homeassistant can't access the sigenstor. For one of the IPs (.50.17) I get a request timeout, and with the other (.50.220) I get a destination host unreachable.

User @micvppl can ping his sigenstor from his homeassistant instance.
![384623574-00743e8f-0b3b-46d1-ad75-2a56ba19bd7f](https://github.com/user-attachments/assets/8c13db60-bf98-4c2d-b3c7-13c3a0e9cd10)

But @TypQxQ can't ping his wired install from his system.
```
C:\Windows\System32>ping 192.168.1.106

Pinging 192.168.1.106 with 32 bytes of data:
Request timed out.
Request timed out.
Request timed out.
Request timed out.

Ping statistics for 192.168.1.106:
    Packets: Sent = 4, Received = 0, Lost = 4 (100% loss),

C:\Windows\System32>
```

I cannot ping the sigenstor from my main network, or from homeassistant on VLAN 50.  However I can ping things on VLAN 50 from everywhere else in the network, and can ping things in the main network from VLAN 50.  So I still don't think it's necessarily a network issue, but I'm fast running out of ideas.

The modbus software linked in the documentations won't download, but I have tried an equivalent Modbus Poll and it won't even connect to the IP.

One thing I tried is logging into the sigen app in the brower and seeing what the cloud API queries.  Interestingly I can ping that IP, however it's not the IP for the modbus server so, as expected, doesn't work with the integration.

Do we know if there are there any other settings in the sigenergy app that have to be enabled to allow this integration to work?  Or does anyone have any other ideas about how to diagnose the issue?

Many thanks in advance,
Matt

**Comment by TypQxQ** [comment]
Looks like the port sure is open but you have something stopping port 502.

Are you sure about the ip address?
Try disconnecting and see if it dissapears from the network map on the router and then reapper when plugged in.

Otherwise check any firewall configurations, are all on wired or mixed wired and wifi? Some routers are sensible.

/Andrei

**Reply by wigglememore** [reply]
I think you're right.

I disconnected and reconnected and now I can ping the IP (no idea why but I can) so that's (I think) good.  I assume then there is something in my unifi controller which is stopping port 502.  I think(?) even though homeassistant and the sigenstor are on the same subnet, things are still routed through the router/firewall; I have vague menories of someone mentioning this about Unifi gear.  I'll make a post on the Ubiquiti forum and see what they say, maybe I need to set up some port forwarding rules.

Everything is wired (homeassistant, sigenstor, PCs etc).

Thanks for your help so far.  We'll get to the bottom of this soon!

**Reply by EnterTheMaelstrom** [reply]
I also have an Unifi system running with HA and Sigenstor on a wired connection, so if I can supply any settings, I would love to help.
I had no problems accessing the Sigenstor Modbus whatsoever, so maybe you have some special routing / firewall settings I don't have. 

You can do a "tracert 192.168.50.17" and maybe see where the signal get's blocked. If it's your firewall it wouldn't even get past your switch (if you have one).

**Comment by yxbh** [comment]
Following this. I have the same issue at the moment. Unifi network with the Sigenstor connected via ethernet. I can ping one of the IPs from HA but the HA integration returns unavailable for entities.

**Comment by TypQxQ** [comment]
Can you try connecting both the HA server and Sigenergy to a unmanaged switch and that in turn to your regular network?
I’m no network expert but having them on a dumb switch could help you bypass any routing.

**Reply by wigglememore** [reply]
This was going to be my next port of call if it didn't get solved, definitely a good idea to avoid any possible routing/firewall issues.

**Comment by wigglememore** [comment, marked as answer]
Thanks for all of the suggestions above.  I spoke to my installer again who, despite the picture I posted at the top of this discussion, had not in fact enabled the modbus server.  As soon as they did, the integration started working as expected.

So unfortuantely, whilst some interesting points were raised that might help others, in this instance, everything about the integration and my network setup was completely fine.

**Reply by yxbh** [reply]
Do you know if your installer just haven't hit the "save" button when they did it or they somehow turned it off again?
